react/Condition.ts

```diff
@@ -27,6 +28,13 @@ const Condition: FC<Props> = ({ children, conditions, match }) => {
     return matches
   }, [conditions, match, subjects, values])
 
+  useEffect(() => {
+    dispatch({
+      type: 'UPDATE_MATCH',
+      payload: { matches: !!matches },
+    })
```

Comment:
What is the purpose of this?

Reply:
Doesn't the provider knows if it matched or not?

Reply:
This is used for the `condition.default` to know if it should be rendered or not. Without this PR, only each `condition` block knows if they matched or not.

Reply:
I see...

that's weird isnt it?

What if this component was like:

```json
{
  "condition-layout.product": {
    "blocks": ["condition.default"],
    "props": {
      "conditions": []
    },
    "children": [
       "rich-text"
     ]
  }
}
```
Then the component it's self sufficient to decide which block it will render.

Reply:
At the time we wrote the RFC, it was not possible to mix blocks and children

Reply:
🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 🤔 

Less switch-y, more if-y? Can do! Less blocks is always better.

Reply:
I've just implemented what @klzns suggested and hit a wall. If we put together the `condition-layout` and `condition` blocks, we wouldn't be able to have a "default" case for multiple conditions:

Imagine a PDP with the following conditions:

`if id === 12 then display X`
`if id === 20 then display Y`
`else display Z`